parity-clib/src/lib.rs

```diff
@@ -87,19 +92,42 @@ pub unsafe extern fn parity_config_destroy(cfg: *mut c_void) {
 }
 
 #[no_mangle]
-pub unsafe extern fn parity_start(cfg: *const ParityParams, output: *mut *mut c_void) -> c_int {
+pub unsafe extern fn parity_start(cfg: *const ParityParams, logger: Logger, output: *mut *mut c_void) -> c_int {
 	panic::catch_unwind(|| {
 		*output = ptr::null_mut();
 		let cfg: &ParityParams = &*cfg;
 
+		let mode = {
+			if logger.mode_len == 0 {
```

Comment:
Make sure no needless allocation is performed but after looking this up neither `slice::from_raw_parts` or `String::from_utf8` does allocate remove this!

Reply:
We're not using this check in `parity_config_from_cli`, I think it's safe since `mem::align_of::<u8> == 1` and also `String::from_utf8(slice.to_owned())` does not allocate if the `slice.is_empty()`.

Reply:
Yeah, it is redundant thus it should be removed

However, when it comes to log-file we need the check because an empty string as `file name` will be a run-time error/panic